src/index.ts

```diff
@@ -0,0 +1,27 @@
+import fp from 'fastify-plugin'
+import { flashFactory, ExtendedRequest } from './flash'
+
+export = fp(
+  function(fastify, opts = {}, done) {
+    const flash = flashFactory()
+
+    fastify.decorateRequest('flash', flash.request)
+    fastify.decorateReply('flash', flash.reply)
+
+    fastify.addHook('preHandler', function(request: ExtendedRequest, reply, next) {
```

Comment:
Why not use `onRequest`?

Reply:
Because of `fastify-session` dependency. We could change this hook if we stick with `fastify-secure-session`

Reply:
Ok for me, @mcollina?